README.md

```diff
@@ -1,79 +1,91 @@
+- Run the script `build_xcframework.sh`, located in the project root.  
+  - After execution, the project’s **build** folder will contain the file **nfclib.xcframework**.  
 
-Nüüd on nfc teek rakendusse integreeritud.
+### Add the Library to the Application
+- Open the project where you want to integrate the `nfclib` library.  
+- Select the project, then under **TARGETS**, choose the correct target.  
+- In the **General** tab of the target, find the **Frameworks and Libraries** section.  
+- Click **+ → Add Other… → Add Files… → Select nfclib.xcframework**.  
 
-# Teegi liidesed id-kaardiga suhtluseks
-Kõik avalikud operatsioonid on kirjeldatud `CardOperations` protokollis.
+The NFC library is now integrated into your application.  
 
-Järgnevalt on nimetatud operatsioonid, mida teek võimaldab.
+# Library Interfaces for ID Card Communication
+All public operations are defined in the `CardOperations` protocol.  
 
-Tagastab, kas NFC on seadmel toetatud.
-`public func isNFCSupported() -> Bool`
+The following operations are provided by the library:  
 
-Loeb asünkroonselt kaardilt avalikku teavet kaardi omaniku kohta
-`public func readPublicInfo(CAN: String) async throws -> CardInfo`
+Returns whether NFC is supported on the device:  
+```swift
+public func isNFCSupported() -> Bool
+```
 
-Loeb asünkroonselt kaardilt autentimise sertifikaadi.
-`public func readAuthenticationCertificate(CAN: String) async throws -> SecCertificate`
+Asynchronously reads public information about the cardholder from the card:
+```swift
+public func readPublicInfo(CAN: String) async throws -> CardInfo`
+```
 
-Loeb asünkroonselt kaardilt allkirjastamise sertifikaadi.
-`public func readSigningCertificate(CAN: String) async throws -> SecCertificate`
+Asynchronously reads the authentication certificate from the card:
+```swift
+public func readAuthenticationCertificate(CAN: String) async throws -> SecCertificate`
+```
 
-Hangib andmeid WebEID autentimiseks, kasutades antud volikirju ja väljakutset.
-`public func loadWebEIDAuthenticationData(CAN: String, pin1: String, challenge: String, origin: String) async throws -> WebEidData`
+Asynchronously reads the signing certificate from the card:
+```swift
+public func readSigningCertificate(CAN: String) async throws -> SecCertificate`
+```
 
-Viib läbi allkirjastamise operatsiooni, kasutades eelnevalt arvutatud räsi (toetatud on ainult SHA-384) ja PIN-koodi
-`public func sign(CAN: String, hash: Data, pin2: String) async throws -> Data`
+Retrieves data required for WebEID authentication, using the provided credentials and challenge:
+```swift
+public func loadWebEIDAuthenticationData(CAN: String, pin1: String, challenge: String, origin: String) async throws -> WebEidData`
+```
+
+Performs a signing operation using a precomputed hash (only SHA-384 supported) and the PIN2 code:
+```swift
+public func sign(CAN: String, hash: Data, pin2: String) async throws -> Data`
+```
```

Comment:
```suggestion
```

Unblocks PIN1 using the PUK code and sets a new PIN:
```swift
public func unblockPin1(CAN: String, puk: String, newCode: String) async throws
```

Unblocks PIN2 using the PUK code and sets a new PIN:
```swift
public func unblockPin2(CAN: String, puk: String, newCode: String) async throws
```
```